packages/techdocs-common/src/stages/publish/local.ts

```diff
@@ -102,7 +107,9 @@ export class LocalPublish implements PublisherBase {
           .then(response =>
             response
               .json()
-              .then(techdocsMetadataJson => resolve(techdocsMetadataJson))
+              .then(techdocsMetadataJson =>
+                resolve(JSON.parse(techdocsMetadataJson)),
```

Comment:
Unfortunately, this throws an error.

`techdocsMetadataJson` is the content of the file `techdocs_metadata.json`. So, `techdocsMetadataJson` is equal to the following string

```
"{ site_name: 'example-docs', site_description: 'None' }"
```

`JSON.parse()` on the string above fails with Syntax error and unexpected token `s`. because the keys in the JSON are not in quotes.

A valid JSON string for `JSON.parse()` looks like this (taken from your test cases) 😄 
```
{ "site_name": "backstage", "site_description": "site_content" }
```

So, we need a better `JSON.parse` which is not as strict and can parse JSON strings without quotes. Looks like [JSON5](https://github.com/json5/json5) is something we need. (It's already a dependency in `yarn.lock` and looks reliable).

Reply:
Thanks for bringing this up! It completely eluded me and at the time I had not tested out how it all looks, lesson learned 😂 

To fix these issues I took a slightly different approach.

For `local` I noticed we're using `.json()` on the response; I figured this would return a JSON string but actually it parses the response and [returns a JS object (or primitive) representation](https://developer.mozilla.org/en-US/docs/Web/API/Body/json#return_value). To adapt I simply trimmed the argument name (to `techdocsMetadata`). 

_Note:_ we are using `json()` in `techdocs` frontend plugin too so I removed the parsing step there too.

For `aws` and `google storage` publishers I took your suggestion and used [JSON5](https://github.com/json5/json5). I added it as a dep. to `techdocs-common` (haven't committed my yarn.lock) and added an additional test for `aws` to check we can parse both singled quote encoded fields as well as double quotes.

During debugging I noticed for the service we were logging the object instead of the name (`entityName`). Changed to `entityName.namespace/entityName.name`.

Please find attached below a screenshot of the frontend spun up locally. Looking at the `backstage` techdocs component the header now renders correctly along with the description. Looking at the networking call to `/api/techdocs/metadata/techdocs/<ns>/<kind>/backstage` the response payload is our JSON encoded metadata 

![image](https://user-images.githubusercontent.com/18265670/104757002-6b0fd080-5754-11eb-91de-2b891b0a5c09.png)